drivers/meross-smart-wifi-garage-door-opener.groovy

```diff
@@ -51,50 +53,58 @@ def getDriverVersion() {
 }
 
 def sendCommand(int open) {
-    if (!settings.messageId || !settings.deviceIp || !settings.sign || !settings.timestamp) {
+    def currentVersion = device.currentState('version').value.replace(".","").toInteger()
+
+    // Firmware version 3.2.3 and greater require different data for request
+    if (!settings.deviceIp || !settings.uuid || (currentVersion >= 323 && !settings.key) || (currentVersion < 323 && (!settings.messageId || !settings.sign || !settings.timestamp))) {
         sendEvent(name: 'door', value: 'unknown', isStateChange: false)
-        log 'missing setting configuration'
+        log.warn('missing setting configuration')
         return
     }
-
     sendEvent(name: 'door', value: open ? 'opening' : 'closing', isStateChange: true)
 
     try {
-        log 'do sendCommand'
+        def payloadData = currentVersion >= 323 ? getSign() : [MessageId: settings.messageId, Sign: settings.sign, CurrentTime: settings.timestamp]
```

Comment:
what is getSign calling?

Reply:
I committed the wrong project.  The sign method is there and it also checks for null values when getting the version number.